metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -512,15 +513,14 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
         discovered_datasets = discovered_tables + discovered_views
 
         if self.config.use_queries_v2:
-            self.report.set_ingestion_stage("*", "View Parsing")
-            assert self.aggregator is not None
+            self.report.set_ingestion_stage("*", VIEW_PARSING)
             yield from auto_workunit(self.aggregator.gen_metadata())
```

Comment:
mostly for my own understanding - `self.aggregator` is only used for view parsing? and then there's another aggregator within `SnowflakeQueriesExtractor`?

if so, should the `self.aggregator.gen_metadata()` call happen first and then lineage generation happen afterwards?

Reply:
> mostly for my own understanding - self.aggregator is only used for view parsing? and then there's another aggregator within SnowflakeQueriesExtractor?

Yes, when `use_queries_v2 is true`

When `use_queries_v2 is false`:
 - and `include_table_lineage is false` -> then self.aggregator is used for only view lineage
 - and `include_table_lineage is true` -> then self.aggregator is used for both view and table lineage.